native/src/seal/polyarray.h

```diff
@@ -0,0 +1,206 @@
+#pragma once
+
+#include "seal/context.h"
+#include "seal/ciphertext.h"
+#include "seal/memorymanager.h"
+#include "seal/publickey.h"
+#include <stdexcept>
+
+using namespace seal::util;
+
+namespace seal
+{
+
+        /**
+        A class holding an RNS based polynomial array that can only be loaded
+        with data once. This class keeps a copy of the coefficient modulus
+        associated with the array so that it can be used to convert back and
+        forth between RNS and multiprecision form.
+        */
+        class StaticPolynomialArray
```

Comment:
I would just rename this class to RingPolynomial.

Reply:
I've renamed it `PolynomialArray` to reflect than 1) it contains more than one polynomial and 2) this class doesn't do any "ring-y" things since it is essentially just a container.